plugins/lua/buildingplan/planneroverlay.lua

```diff
@@ -368,8 +370,44 @@ function ItemLine:has_filter()
 end
 
 function ItemLine:get_filter_text()
-    -- TODO: make this show the filter's materials instead of "edit filters"
-    return self:has_filter() and '[edit filters]' or '[any material]'
+    local buildingplan = require('plugins.buildingplan')
+    if not buildingplan.hasFilter(
+        uibs.building_type, uibs.building_subtype, uibs.custom_type, self.idx-1)
+    then
+        return '[any material]'
+    end
+    local mats = buildingplan.getMaterialFilter(
+        uibs.building_type, uibs.building_subtype, uibs.custom_type, self.idx-1)
+    local cats = buildingplan.getMaterialMaskFilter(
+        uibs.building_type, uibs.building_subtype, uibs.custom_type, self.idx-1)
+    local enabled_mat_names = {}
+    local enabled_cat_names = {}
+    for name,props in pairs(mats) do
+        local enabled = props.enabled == 'true' and cats[props.category]
+        if enabled then table.insert(enabled_mat_names, name) end
+    end
+    if #enabled_mat_names == 1 then
+        local mat = enabled_mat_names[1]
+        if mat:len() < 17 then
```

Comment:
17 should be a constant so it can be shared between the field width+2 and this check